benchmarks/oscar/ootb/oscar_parser.py

```diff
@@ -0,0 +1,87 @@
+import re
+import csv
+import argparse
+import sys
+from collections import OrderedDict
+
+
+class OscarLogParser:
+    def __init__(self, select_epoch):
+        super(OscarLogParser, self).__init__()
+        self.select_epoch = select_epoch
+
+    def parse(self, path):
+        print(f"Parsing {path}")
+        train_time_regex = re.compile(".*Train Time: (?P<train_time>[0-9]*\.[0-9]*).*")
+        global_step_regex = re.compile(
+            ".*Epoch: (?P<epoch>[0-9]*), global_step: (?P<global_step>[0-9]*).*"
+        )
+        global_batch_size_regex = re.compile(
+            ".*Total train batch size.* = (?P<global_batch_size>[0-9]*).*"
+        )
+
+        train_times = []
+        global_steps = []
+        global_batch_size = None
+
+        with open(file=path, mode="r", encoding="UTF-8") as f:
+            for line in f:
+                match = train_time_regex.match(line)
+                if match:
+                    train_time = float(match["train_time"])
+                    train_times.append(train_time)
+
+                match = global_step_regex.match(line)
+                if match:
+                    global_step = float(match["global_step"])
+                    global_steps.append(global_step)
+
+                match = global_batch_size_regex.match(line)
+                if match:
+                    global_batch_size = int(match["global_batch_size"])
+
+        metrics = OrderedDict()
+        if self.select_epoch >= len(global_steps):
+            print(
+                f" the bigest epoch is {len(global_steps) - 1}, your select epoch is {self.select_epoch}, please select a smaller epoch."
+            )
+            sys.exit(0)
+        if len(global_steps) > 0 and len(train_times) > 0:
+            metrics["performance_step"] = self.select_epoch
+            metrics["performance(it/s)"] = global_steps[self.select_epoch] / sum(
+                train_times[: self.select_epoch + 1]
+            )
+            metrics["performance(samples/s)"] = metrics["performance(it/s)"] * global_batch_size
```

Comment:
We were previously doing
```
train_time=$(grep -Eo "Train Time: [0-9\.]*" log.txt | sed 's/Train Time: //g' | head -n1)
global_step=$(grep  "global_step = [0-9]*, average loss = " log.txt | grep -Eo "global_step = [0-9]*" | sed "s/global_step = //g")
val=$(echo "scale=4 ; $global_step / $train_time" | bc)
```

Reply:
We can discuss about how we pick numbers to calculate the metrics offline, thanks.